lychee-lib/src/helpers/url.rs

```diff
@@ -0,0 +1,85 @@
+use linkify::LinkFinder;
+
+/// Remove all GET parameters from a URL.
+/// The link is not a URL but a String as it may not have a base domain.
+pub(crate) fn remove_get_params_and_fragment(url: &str) -> &str {
+    let path = match url.split_once('#') {
+        Some((path_without_fragment, _fragment)) => path_without_fragment,
+        None => url,
+    };
+    let path = match path.split_once('?') {
+        Some((path_without_params, _params)) => path_without_params,
+        None => path,
+    };
+    path
```

Comment:
It would be smart to do the query string split first, so that the dedicated fragment split can be skipped in case. Would something like this work?
```rust
    let path = match path.split_once('?') {
        Some((path_without_params, _params)) => path_without_params,
        None => match url.split_once('#') {
            Some((path_without_fragment, _fragment)) => path_without_fragment,
            None => url,
        };
    };
```
- So if there is a query string, strip and return, as the fragment has been stripped off together with the query string.
- Only if there is no query string, strip off the fragment.

Reply:
Makes sense. I changed that now. Thanks!

Reply:
What about questionmarks in anchors/fragments?

Reply:
> What about questionmarks in anchors/fragments?

Indeed I also though about that, see my edit. I think it is very care, but not worth it cause false negatives for such a little performance benefit. But as noted, best is probably to remove fragments from all URLs (as long as they are not verified) and e.g. add them in a different way to the output to not give a false impression that they would be verified: https://github.com/lycheeverse/lychee/pull/262#issuecomment-914297175

Reply:
Reverted the change and added a test. It never crossed my mind that fragments could contain question marks. Wow.